lib/Dialect/Linalg/Transforms/LinalgTransforms.cpp

```diff
@@ -106,3 +111,38 @@ bool mlir::linalg::detail::isProducedByOpOfTypeImpl(
   }
   return false;
 }
+
+LogicalResult
+mlir::linalg::permuteGenericLinalgOp(PatternRewriter &rewriter, Operation *op,
+                                     ArrayRef<unsigned> permutation,
+                                     StringRef linalgMarker) {
+  // If permutation is empty, there is nothing to be done.
+  if (permutation.size() == 0)
+    return failure();
+
+  OpBuilder::InsertionGuard g(rewriter);
+  auto linOp = cast<LinalgOp>(op);
+  rewriter.setInsertionPoint(linOp);
+  ScopedContext scope(rewriter, linOp.getLoc());
```

Comment:
I cannot see offhand why you need this?

Reply:
We didn't....Removed. Thanks for noticing.